### PR TITLE
refactor zip validation + smarter default radius

### PR DESCRIPTION
**Pull request by cathydeng**
- validate zip before proceeding to next step
  - ultimately decided to keep the state for whether zip is valid outside of ZipInput, since we need it to control button behavior and ZipInput does not handle the submit button. but open to ideas on neater ways to organize this
- set a smaller default radius when searching a dense zip

https://user-images.githubusercontent.com/1406537/177466678-e9faf706-b473-4dd9-8e85-298bec1d2d22.mov

### CODE REVIEW

File: src/util.tsx [hunk 1]
```diff
@@ -13,10 +13,13 @@ import {
   ZipData,
   TypeOfHelp,
   FeePreference,
+  ZipSearchMetadata,
 } from "./types";
 import coloradoZipData from "./data/colorado_zip_data.json";
 
 export const DEFAULT_RADIUS_MILES = 10;
+export const DEFAULT_DENSE_RADIUS_MILES = 5;
+export const DENSITY_CUTOFF_POP_PER_SQ_MI = 1000;
```
Comment (cathydeng) [inline src/util.tsx:22]: picked this somewhat arbitrarily based on spot checking a buncha zips w/ varying densities to see how many results there were

File: src/types.ts [hunk 2]
```diff
@@ -138,7 +138,6 @@ export type CareProviderSearchResult = CareProvider &
 
 export type SearchResult = {
   results: CareProviderSearchResult[];
-  error: string | null;
```
Comment (cathydeng) [inline src/types.ts:141]: keeping the object instead of just returning results b/c we'll need to add more metadata anyways when we do pagination

**Review by jlhogan — COMMENTED** [review]
I think passing in state for zip validity to ZipInout is fine. Can't think of a simpler way since we need to do context-specific logic around how to handle valid/invalid state. Do think it would be good to consolidate zip checking logic either by passing zip search meta to getMatchingCare (a context? So it's available everywhere?) or at least calling getZipMeta again in getMatchingCare

File: src/util.tsx [hunk 3]
```diff
@@ -119,17 +122,32 @@ export const meetsFeePreferences = (
...
+
 // TODO: figure out how to limit results if there are too many
 export function getMatchingCare(
   careData: CareProvider[],
   filters: SearchFilters
 ): SearchResult {
   const { zip, miles: milesStr, typesOfHelp, feePreferences } = filters;
-  const miles = parseInt(milesStr);
+  const miles = (milesStr && parseInt(milesStr)) || DEFAULT_RADIUS_MILES;
   if (zip.length !== 5) {
```
[the remark this answers is not in the document]
Comment (cathydeng) [reply src/util.tsx:148]: yeah agreed it’s awk, i think i’ll tweak it a bit and call getZipMetadata in here

**Review by jlhogan — APPROVED** [review]
🔆

**Pull request merged**